File: src/main/kotlin/com/sigma/agar/utils/protocol/Protocol.java
package com.sigma.agar.utils.protocol;

import com.sigma.agar.physics.entity.Thing;
import com.sigma.agar.utils.Tuple;
import org.web.httpserver.Session;

import java.util.List;

public interface Protocol {
	void dispatch(Tuple center, float width, float height, List<Thing> entities, Session session);
}

File: src/main/kotlin/com/sigma/agar/utils/protocol/Protocol.kt
package com.sigma.agar.utils.protocol

import com.sigma.agar.physics.things.Thing
import com.sigma.agar.utils.Tuple
import org.webutil.httpserver.Session

interface Protocol {
    fun dispatch(center: Tuple, width: Float, height: Float, things: List<Thing>, end_time_ms: Long, session: Session)
}